### PR TITLE
⬆️ update hapi dependencies

### PR DESCRIPTION
**Pull request by eliascodes**
Major bump to v18 was accompanied by a package namespace change on all hapi-related packages to `@hapi/PACKAGE_NAME`. This has inflated the number of files touched by this PR considerably.

### Changes
- Remove the now-incompatible `hapi-qs`, replace with custom implementation using `qs` and new Hapi API
- add simple wrapper around arguments to `server.inject` to accommodate changes to the `server.inject` API

#### api
* @types/qs: "^6.5.3",
* @types/ramda: "^0.26.12",
* boom: "^7.4.2"
* hapi: "^18.3.1"
* hapi-auth-jwt2: "^8.6.1",
* joi: "^15.1.0"

#### dashboard
* joi: "^15.1.0"

### CODE REVIEW

File: api/src/api/v1/users/volunteers/__tests__/volunteer_logs.test.integration.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ describe('API /users/me/volunteer-logs', () => {
       const res = await server.inject({
```
Comment (astroash) [inline api/src/api/v1/users/volunteers/__tests__/volunteer_logs.test.integration.ts]: How come these tests aren't using the `injectCfg` wrapper?

File: api/src/auth/strategies/standard/validate.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const StandardCredentials: TCredentials = {
   },
 
   fromRequest (request: Hapi.Request) {
-    return { ...request.auth.credentials.user, scope: request.auth.credentials.scope };
+    return Object.assign({ scope: request.auth.credentials.scope }, request.auth.credentials.user);
```
Comment (astroash) [inline api/src/auth/strategies/standard/validate.ts]: Isn't the spread and `Object.assign` doing the same thing?

Comment (eliascodes) [reply api/src/auth/strategies/standard/validate.ts]: Somehow TS loses typings if I use spread here

File: api/src/api/v1/users/volunteers/__tests__/volunteer_logs.test.integration.ts (hunk 1 above)
Comment (eliascodes) [reply api/src/api/v1/users/volunteers/__tests__/volunteer_logs.test.integration.ts]: I did those first before I wrote the wrapper

Comment (astroash) [reply api/src/api/v1/users/volunteers/__tests__/volunteer_logs.test.integration.ts]: Can we wrap these for consistency?

**Comment by astroash** [comment]
@eliascodes can the tests fix commit be cherry picked to a separate PR? Just incase we do have issues with it in the future it will be buried under this PRs title

**Pull request merged**